tests/test_updown.py

```diff
@@ -4,144 +4,243 @@
 import tempfile
 from flask import url_for
 
+from conftest import webdav_server
+
+
+@pytest.mark.timeout(30)
+def test_health(app: Any, client: Any):
+    with webdav_server():
+        with app.app_context():
+            print(url_for('health'))
+            r = client.get(url_for('health'))
+            print(r.json, url_for('health'))
+            assert r.status_code == 200
+
+
+@pytest.mark.timeout(30)
+def test_list(app: Any, client: Any):
+    with webdav_server():
+        with app.app_context():
+            r = client.get(url_for('list', path="lst"))
+            assert r.status_code == 200
+            print(r.json)
+
+
+@pytest.mark.timeout(30)
+def test_download(app: Any, client: Any):
+    with webdav_server() as server:
+        subprocess.check_call([
+            "dd", "if=/dev/random",
+            f"of={server['config']['provider_mapping']['/']}/md5sum-lst.txt",
+            "bs=1M", "count=10"
+        ])
+        with app.app_context():
+            r = client.get(url_for('fetch', path="md5sum-lst.txt"))
+            assert r.status_code == 200
+            print(r.json)
+
+
+@pytest.mark.timeout(30)
+def test_webdav_list(app: Any, client: Any):
+    with webdav_server():
+        with app.app_context():
+            r = client.open(url_for('webdav', path="lst"), method='PROPFIND')
+            assert r.status_code in [200, 207]
```

Comment:
also here, could you check that it lists what we want? at least some of the output should be validated